TombEngine/Objects/TR1/Entity/tr1_big_rat.cpp

```diff
@@ -83,21 +83,24 @@ namespace TEN::Entities::Creatures::TR1
 
 	bool IsRatOnWater(ItemInfo* item)
 	{
-		auto* creature = GetCreatureInfo(item);
-		auto* object = &Objects[item->ObjectNumber];
-
 		int waterDepth = GetWaterSurface(item->Pose.Position.x, item->Pose.Position.y, item->Pose.Position.z, item->RoomNumber);
-		if (waterDepth != NO_HEIGHT)
-		{
-			creature->LOT.Step = BLOCK(20);
-			creature->LOT.Drop = -BLOCK(20);
-		}
-		else
+		
+		if (item->IsCreature())
 		{
-			creature->LOT.Step = CLICK(1);
-			creature->LOT.Drop = -CLICK(1);
-		}
+			auto* creature = GetCreatureInfo(item);
 
+			if (waterDepth != NO_HEIGHT)
+			{
+				creature->LOT.Step = BLOCK(20);
```

Comment:
This is not related to PR itself, but it greatly confuses me that function which is supposed to check if rat is on water, also modifies creature LOT. I would rename `IsRatOnWater` to just `RatOnWater`.

Reply:
That's what allows it to climb up and down the blocks right? 

I can change the name in a quick commitment.

Reply:
Yes please :)

Reply:
Yes, I've already done.